rotkehlchen/chain/base/modules/efp/decoder.py

```diff
@@ -0,0 +1,94 @@
+UPDATE_LIST_STORAGE_LOCATION: Final = b'\xe6PM\xf8\xb7bT\xe82m\xba\x1d\x1f\xd1\xacI6\xde(\xca\xf5e\xb9\xc5\xa1\xad\x0f\x10\xd6\xc9\x8e\xc2'  # noqa: E501
+
+
+class EfpDecoder(EfpCommonDecoder):
+
+    def __init__(
+            self,
+            base_inquirer: 'BaseInquirer',
+            base_tools: 'BaseDecoderTools',
+            msg_aggregator: 'MessagesAggregator',
+    ) -> None:
+        super().__init__(
+            evm_inquirer=base_inquirer,
+            base_tools=base_tools,
+            msg_aggregator=msg_aggregator,
+            list_records_contract=string_to_evm_address('0x41Aa48Ef3c0446b46a5b1cc6337FF3d3716E2A33'),
+        )
+
+    def _decode_account_metadata_events(self, context: DecoderContext) -> DecodingOutput:
+        """Decode events from the EFPListRegistry contract."""
+        if context.tx_log.topics[0] != UPDATE_ACCOUNT_METADATA:
+            return DEFAULT_DECODING_OUTPUT
+
+        key = context.tx_log.data[96:128].rstrip(b'\x00').decode()
+        if key != 'primary-list':
+            return DEFAULT_DECODING_OUTPUT
+
+        return DecodingOutput(event=self.base.make_event_from_transaction(
+            transaction=context.transaction,
+            tx_log=context.tx_log,
+            event_type=HistoryEventType.INFORMATIONAL,
+            event_subtype=HistoryEventSubType.NONE,
+            asset=A_ETH,
+            balance=Balance(),
+            location_label=(user_address := bytes_to_address(context.tx_log.topics[1])),
+            notes=f'Create EFP primary list for {user_address}',
+            counterparty=CPT_EFP,
+        ))
+
+    def _decode_list_registry_events(self, context: DecoderContext) -> DecodingOutput:
+        """Decode events from the EFPListRegistry contract."""
+        if context.tx_log.topics[0] != ERC20_OR_ERC721_TRANSFER:
+            return DEFAULT_DECODING_OUTPUT
+
+        return DecodingOutput(action_items=[ActionItem(
+            action='transform',
+            from_event_type=HistoryEventType.RECEIVE,
+            from_event_subtype=HistoryEventSubType.NONE,
+            location_label=(user_address := bytes_to_address(context.tx_log.topics[2])),
+            asset=Asset('eip155:8453/erc721:0x0E688f5DCa4a0a4729946ACbC44C792341714e08'),  # EFP List NFT  # noqa: E501
+            address=ZERO_ADDRESS,
+            to_notes=f'Receive EFP list NFT for {user_address}',
+            to_counterparty=CPT_EFP,
+        )])
+
+    # -- DecoderInterface methods
+
+    def addresses_to_decoders(self) -> dict['ChecksumEvmAddress', tuple[Any, ...]]:
+        return {
+            self.list_records_contract: (self._decode_list_op_events,),
+            string_to_evm_address('0x5289fE5daBC021D02FDDf23d4a4DF96F4E0F17EF'): (self._decode_account_metadata_events,),  # EFPAccountMetadata  # noqa: E501
+            string_to_evm_address('0x0E688f5DCa4a0a4729946ACbC44C792341714e08'): (self._decode_list_registry_events,),  # EFPListRegistry  # noqa: E501
+        }
```

Comment:
This is a child class. do not repeat the super class code.

```suggestion
        return super().addresses_to_decoders() | {
            string_to_evm_address('0x5289fE5daBC021D02FDDf23d4a4DF96F4E0F17EF'): (self._decode_account_metadata_events,),  # EFPAccountMetadata  # noqa: E501
            string_to_evm_address('0x0E688f5DCa4a0a4729946ACbC44C792341714e08'): (self._decode_list_registry_events,),  # EFPListRegistry  # noqa: E501
        }
```